Observer/App/Action/Action.php

```diff
@@ -0,0 +1,108 @@
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Tagging\Observer\App\Action;
+
+use Magento\Framework\App\Response\Http as HttpResponse;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Exception\InputException;
+use Magento\Framework\Stdlib\Cookie\CookieSizeLimitReachedException;
+use Magento\Framework\Stdlib\Cookie\FailureToSendException;
+use Magento\Framework\Stdlib\CookieManagerInterface;
+use Nosto\Tagging\Model\Customer\Customer;
+use Magento\Framework\Stdlib\Cookie\PublicCookieMetadata;
+
+class Action implements ObserverInterface
+{
+    /**
+     * @var HttpResponse $response
+     */
+    private $response;
+    /**
+     * @var CookieManagerInterface
+     */
+    private $cookieManager;
+
+    /**
+     * Action constructor.
+     * @param HttpResponse $response
+     * @param CookieManagerInterface $cookieManager
+     */
+    public function __construct(
+        HttpResponse $response,
+        CookieManagerInterface $cookieManager
+    ) {
+        $this->response = $response;
+        $this->cookieManager = $cookieManager;
+    }
+
+    /**
+     * @param Observer $observer
+     * @throws InputException
+     * @throws CookieSizeLimitReachedException
+     * @throws FailureToSendException
+     */
+    public function execute(Observer $observer) // @codingStandardsIgnoreLine
+    {
+        $cookieValue = $this->cookieManager->getCookie(Customer::COOKIE_NAME);
+        $cookieValueHttp = $this->cookieManager->getCookie(Customer::HTTP_COOKIE_NAME);
+
+        if (empty($cookieValue) || ($cookieValue === $cookieValueHttp)) {
```

Comment:
feels like missing this case
```php
if (empty($cookieValue) && !empty($cookieValueHttp) {
 // set the $cookieValue
}
```

Reply:
The `2c.cId` is set via Nosto Script. It should only be empty if Webkit deletes the cookie and the user visits the page for the first time.
Or am I missing something?

Reply:
I mean the fist request after 7 days from previous interaction, before script had a chance to set the value, `2c.cId.http` cookie is still present, `2c.cId` is not

Reply:
Gotcha